### PR TITLE
Geometry Color Tests More

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Adding more details on geometry coloring tests

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

**Comment by QilongTang** [comment]
Still caching up on geometry coloring tests, need to add something for transparency soon

File: src/VisualizationTests/HelixWatch3DViewModelTests.cs [hunk 1]
```diff
@@ -724,24 +724,41 @@ public void Python_CreatesVisualizations()
...
 
+            Assert.AreEqual(BackgroundPreviewGeometry.Count(), 4);
             Assert.True(BackgroundPreviewGeometry.HasAnyColorMappedMeshes());
+
+            // These checks are more specific to this test
+            // Expecting 6 color definitions for vertices in the DynamoGeometry
+            var dynGeometry = BackgroundPreviewGeometry.OfType<DynamoGeometryModel3D>().FirstOrDefault();
+            var numberOfColors = dynGeometry.Geometry.Colors.Count;
+            Assert.AreEqual(numberOfColors, 6);
+            Assert.AreEqual(((PhongMaterial)dynGeometry.Material).DiffuseMap.Width, 52);
```
Comment (mjkkirschner) [inline src/VisualizationTests/HelixWatch3DViewModelTests.cs]: might be good to also check that the colors match the map (or nearly do)

Comment (QilongTang) [reply src/VisualizationTests/HelixWatch3DViewModelTests.cs]: I wanted to but I don't think colors on `DiffuseMap` is exposed from debugging

**Comment by reddyashish** [comment]
Looks good @QilongTang.

File: src/VisualizationTests/HelixWatch3DViewModelTests.cs [hunk 2]
```diff
@@ -721,27 +721,59 @@ public void Python_CreatesVisualizations()
         }
 
         [Test]
-        public void Display_ByGeometryColor_HasColoredMesh()
+        public void Display_FrozenNode_HasTransparentMesh()
         {
             OpenVisualizationTest("Display.ByGeometryColor.dyn");
+            RunCurrentModel();
+            Assert.AreEqual(BackgroundPreviewGeometry.Count(), 4);
```
Comment (aparajit-pratap) [inline src/VisualizationTests/HelixWatch3DViewModelTests.cs]: @QilongTang sorry for the late review but you should swap the arguments to `Assert.AreEqual` in all these cases as the first arg is always the expected value and the second is actual. If a test fails NUnit can report confusing failure messages due to this mistake.

**Pull request merged**